assets/js/admin/editor-lifecycle.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * WordPress dependencies
+ */
+import { subscribe, select } from '@wordpress/data';
+import { store as editorStore } from '@wordpress/editor';
+
+/**
+ * Helper function to fire callbacks on editor lifecycles.
+ *
+ * @param {Object}   options
+ * @param {Function} options.subscribeListener Callback called everytime the subscribe listener is called.
+ * @param {Function} options.onSetDirty        Callback called when the editor becomes dirty.
+ * @param {Function} options.onSaveStart       Callback called when editor starts saving.
+ * @param {Function} options.onSave            Callback called when a save is completed.
+ *
+ * @return {Function} Unsubscribe function.
+ */
+const editorLifecycle = ( {
+	subscribeListener = () => {},
+	onSetDirty = () => {},
+	onSaveStart = () => {},
+	onSave = () => {},
+} ) => {
+	const coreEditorSelector = select( editorStore );
+	let wasSaving = false;
+	let wasDirty = false;
+	let isNew    = false;
```

Comment:
The `editorLifecycle` was borrowed from Sensei LMS, just adding the extra `isNew` variable to check if it's saving a new post or updating an existing one.

With this logic, we continue considering it as new even after an autosave.

Reply:
Approach changed.